File: frontend/src/types/operation-log.ts
export type OperationLog = {
  id: number;
  user_id: number | null;
  username: string | null;
  action: string;
  target_type: string | null;
  target_id: number | null;
  project_id: number | null;
  center_id: number | null;
  ip_address: string | null;
  user_agent: string | null;
  detail_json: Record<string, unknown> | null;
  created_at: string;
};

export type OperationLogList = {
  items: OperationLog[];
  total: number;
  limit: number;
  offset: number;
};

export type OperationLogFilters = {
  user_id?: number;
  username?: string;
  action?: string;
  target_type?: string;
  target_id?: number;
  project_id?: number;
  center_id?: number;
  created_from?: string;
  created_to?: string;
  limit?: number;
  offset?: number;
};

File: backend/app/schemas/operation_log.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict, Field


class OperationLogRead(BaseModel):
    id: int
    user_id: int | None = None
    username: str | None = None
    action: str
    target_type: str | None = None
    target_id: int | None = None
    project_id: int | None = None
    center_id: int | None = None
    ip_address: str | None = None
    user_agent: str | None = None
    detail_json: dict[str, Any] | None = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class OperationLogListRead(BaseModel):
    items: list[OperationLogRead] = Field(default_factory=list)
    total: int
    limit: int
    offset: int
